terraform/Makefile

```
@@ -25,7 +23,7 @@ go-mod-tidy-vendor: $(GO_MOD_TIDY_TARGETS)
 # The -compat=1.18 is needed for the openstack provider. The provider uses golang.org/x/mod, which go 1.18 selects
 # as v0.3.0 but go 1.16 selects as v0.4.2.
 $(GO_MOD_TIDY_TARGETS): go-mod-tidy-vendor.%:
-	cd providers/$* && go mod tidy -compat=1.18 && go mod vendor
+	cd providers/$* && go mod edit -go=$(GOVERSION) && go mod tidy -compat=1.18 && go mod vendor
```

Should we also bump the compat version?
```suggestion
	cd providers/$* && go mod edit -go=$(GOVERSION) && go mod tidy -compat=$(GOVERSION) && go mod vendor
```

---

No idea. I was going to ask the guys that did this if it should be bumped. I will find out. I am currently not familiar with what this actually does.